src/patches/dense_patch_computation.cpp

```diff
@@ -529,12 +492,16 @@ void run_through_dense_slices_streamed(
             res.slice_count_++;
 
             for (auto&& i: application_result.followup_instructions)
-                future_instructions.push(i);
+                future_instructions.push_back(i);
         }
         else if (application_result.maybe_error)
         {
             slice_visitor(slice);
-            throw std::runtime_error{application_result.maybe_error->what()};
+            if (instruction.wait_at_most_for == 0)
+                throw std::runtime_error{application_result.maybe_error->what()};
+            instruction.wait_at_most_for--;
+            future_instructions.push_front(instruction);
+            advance_slice(slice, layout);
```

Comment:
Moved here the followup logic to completely remove the use of both error and followup instructions in the application result